### ISSUE TITLE
Minify does not work on templates

### ISSUE THREAD

**Issue by NitroBAY** [jantimon/html-webpack-plugin#901]
Minify does minify generated html but not the part from the template ?
### Config

```
const client = {
    entry: './client/src/index.js',
    output: {
        filename: 'bundle.js',
        path: path.resolve(__dirname, 'client/dist')
    },
    target: 'web',
    plugins: [
        new HtmlWebpackPlugin({
            minify: {
                collapseWhitespace: true,
            },
        }),
    ]
};

```
produces a minified file but not if I use template (I add to the object passed to HtmlWebpackPlugin             `template: 'client/src/index.html',`).
Okay I understand why this does not matter and why this probably won't be a priority for anyone but if this a bug I report it if you don't care I drop it here so next users of this package will understand what's making minify not working. Simplest fix would be to update readme, if it's not already explicit, so one don't think the package is broken. Anyway I just hope saving next people's time not be annoying with meaningless/very-specific-case issues.
Have a nice day/night.

**Comment by jantimon** [comment]
Minifying should be one of the last steps - could you please add some console.log statements to node-modules/html-webpack-plugin/index.js and try to figure out what is causing this issues?

**Comment by NitroBAY** [comment]
Investigation is going on Watson

**Comment by NitroBAY** [comment]
Oh no sry I'm kinda new with webpack and I forgot again that `watch` doesn't watch the config file in webpack. But as I thought it was doing so, each time I was removing the template line I was manually deleting the old generated html file and as it wouldn't appeared I would relaunch manually this time my webpack. So this is why only automatically generated HTML would get minified and not the other as I was always settings back minify's configuration for the template.
Long story short : it was my fault.
Sorry again thanks for your reactivity and it was cool to see how you were using (perfectly) promises.
